sklearn/decomposition/dict_learning.py

```diff
@@ -282,7 +282,7 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
             algorithm,
             regularization=regularization, copy_cov=copy_cov,
             init=init[this_slice] if init is not None else None,
-            max_iter=max_iter)
+            max_iter=max_iter, n_jobs=n_jobs)
```

Comment:
This is not going to work as it would trigger double Parallel pool I believe.

Proper way to do this would be to directly call _sparse_encode if `algorithm` is `threshold, `lars`, `lasso_lars`, and to keep the `Parallel` call otherwise